samples/bluetooth/central_nfc_pairing/src/main.c

```diff
@@ -0,0 +1,637 @@
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing failed conn: %s, reason %d\n", addr, reason);
+
+	k_poll_signal_raise(&pair_signal, 0);
+	bt_set_oob_data_flag(false);
+}
+
+static struct bt_conn_auth_cb conn_auth_callbacks = {
+	.cancel = auth_cancel,
+	.oob_data_request = auth_oob_data_request,
+	.pairing_confirm = pairing_confirm,
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed
+};
+
+static int check_oob_carrier(const struct nfc_tnep_ch_record *ch_record,
+			     const struct nfc_ndef_record_desc **oob_data)
+{
+	const struct nfc_ndef_ch_ac_rec *ac_rec = NULL;
+
+	if (!ch_record->count) {
+		return -EINVAL;
+	}
+
+	for (size_t i = 0; i < ch_record->count; i++) {
+		if (nfc_ndef_le_oob_rec_check(ch_record->carrier[i])) {
+			*oob_data = ch_record->carrier[i];
+		}
+	}
+
+	if (!oob_data) {
+		printk("Connection Handover Requester not supporting OOB BLE\n");
+		return -EINVAL;
+	}
+
+	/* Look for the corresponding Alternative Carrier Record. */
+	for (size_t i = 0; i < ch_record->count; i++) {
+		if (((*oob_data)->id_length == ch_record->ac[i].carrier_data_ref.length) &&
+		    (memcmp((*oob_data)->id,
+			    ch_record->ac[i].carrier_data_ref.data,
+			    (*oob_data)->id_length) == 0)) {
+			ac_rec = &ch_record->ac[i];
+		}
+	}
+
+	if (!ac_rec) {
+		printk("No Alternative Carrier Record for OOB LE carrier\n");
+		return -EINVAL;
+	}
+
+	/* Check carrier state */
+	if ((ac_rec->cps != NFC_AC_CPS_ACTIVE) &&
+	    (ac_rec->cps != NFC_AC_CPS_ACTIVATING)) {
+		printk("LE OBB Carrier inactive\n");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int oob_le_data_handle(const struct nfc_ndef_record_desc *rec,
+			      bool request)
+{
+	int err;
+	const struct nfc_ndef_le_oob_rec_payload_desc *oob;
+	uint8_t desc_buf[NFC_NDEF_LE_OOB_REC_PARSER_BUFF_SIZE];
+	uint32_t desc_buf_len = sizeof(desc_buf);
+
+	err = nfc_ndef_le_oob_rec_parse(rec, desc_buf,
+					&desc_buf_len);
+	if (err) {
+		printk("Error during NDEF LE OOB Record parsing, err: %d.\n",
+			err);
+	}
+
+	oob = (struct nfc_ndef_le_oob_rec_payload_desc *) desc_buf;
+
+	nfc_ndef_le_oob_rec_printout(oob);
+
+	if ((*oob->le_role != NFC_NDEF_LE_OOB_REC_LE_ROLE_PERIPH_ONLY) &&
+	    (*oob->le_role != NFC_NDEF_LE_OOB_REC_LE_ROLE_PERIPH_PREFFERED)) {
+		printk("Unsupported Device LE Role\n");
+		return -EINVAL;
+	}
+
+	if (oob->le_sc_data || oob->tk_value) {
+		bt_set_oob_data_flag(true);
+	}
+
+	if (oob->le_sc_data) {
+		remote_sc_data = *oob->le_sc_data;
+	}
+
+	if (oob->tk_value) {
+		memcpy(remote_tk_value, oob->tk_value, sizeof(remote_tk_value));
+		use_remote_tk = request;
+	}
+
+	/* Remove all scanning filters. */
+	bt_scan_filter_remove_all();
+
+	err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_ADDR, oob->addr);
+	if (err) {
+		printk("Set filter on device address error: %d\n", err);
+	}
+
+	err = bt_scan_filter_enable(BT_SCAN_ADDR_FILTER, false);
+	if (err) {
+		return err;
+	}
+
+	/* This demo doesn't require active scan */
```

Comment:
This comment is being copy-pasted around :P 
Please delete it, we are using active scan

Reply:
Deleted